akka-stream/src/main/boilerplate/akka/stream/scaladsl/ZipLatestWithApply.scala.template

```diff
@@ -0,0 +1,109 @@
+/**
+ * Copyright (C) 2014-2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+package akka.stream.scaladsl
+
+import akka.stream._
+import akka.stream.impl.StreamLayout
+import akka.stream.stage.{ GraphStage, GraphStageLogic, InHandler, OutHandler }
+
+trait ZipLatestWithApply {
+
+  [2..20#/**
+   * Create a new `ZipLatestWith` specialized for 1 inputs.
+   *
+   * @param zipper zipping-function from the input values to the output value
+   */
+  def apply[[#A1#], O](zipper: ([#A1#]) ⇒ O): ZipLatestWith1[[#A1#], O] =
+    new ZipLatestWith1(zipper)
+  #
+
+  ]
+
+}
+
+[2..20#/** `ZipLatestWith` specialized for 1 inputs */
+class ZipLatestWith1[[#A1#], O] (val zipper: ([#A1#]) ⇒ O) extends GraphStage[FanInShape1[[#A1#], O]] {
+  override def initialAttributes = Attributes.name("ZipLatestWith1")
+  override val shape: FanInShape1[[#A1#], O] = new FanInShape1[[#A1#], O]("ZipLatestWith1")
+  def out: Outlet[O] = shape.out
+  [#val in0: Inlet[A1] = shape.in0#
+  ]
+
+  override def createLogic(inheritedAttributes: Attributes): GraphStageLogic = new GraphStageLogic(shape) {
+      // Without this field the completion signalling would take one extra pull
+      var willShutDown = false
+
+      [#var hasValue0 = false#
```

Comment:
I have tried but unfortunately (but also rather logically), OptionVal has a reference type constraint '>: Null', which means that this would propagate to 'ZipLatest'. Do you see an alternative or is there another version that supports value types? I find meta-programming limits the ugliness somewhat 😉

Reply:
In practice all elements going through the stream will be boxed anyway, I thought there were stages where we use that knowledge and do some casting to AnyRef internally, but now I can't find such a stage, so maybe those has been eliminated.

Reply:
ok, changing

Reply:
I think you'd have to do some somewhat uggly `asInstanceOf[AnyRef]` on the elements and use `OptionVal[AnyRef]` but on second thought, maybe it isn't worth it. Let's keep it as is for now.